src/Sensors.cpp

```diff
@@ -61,21 +61,43 @@ bool Sensors::ChooseTarget(TargetingCriteria crit)
 	PROFILE_SCOPED();
 	bool found = false;
 
+	Player* player = dynamic_cast<Player*>(m_owner);
+
+	if(!player)
+		return false;
+
+	Body* currTarget = player->GetCombatTarget();
+
```

Comment:
dynamic_cast is not particularly performant (depending on platform and compiler, it can walk the entire class inheritance tree). It's not a problem when called once per frame, but in game contexts (Pioneer specifically) it's *usually* a sign of bad architecture.

This is not a requirement to merge at all, but I'd recommend that this function be refactored to take in a Body* of the current combat target and return a Body* of the new combat target; the actual assignment to the player's combat target should be performed in the calling function instead.